File: 백준 문제 풀이 모음/class 3/1541.py
import sys

input = sys.stdin.readline().rstrip()

op_list = []
new_input = ''
for i in range(len(input)) :
    if input[i] == '-'  :
        new_input += ' - '
    elif input[i] == '+' :
        new_input += ' + '
    else :
        new_input += input[i]

new_input = new_input.split(' ')


for i in range(1, len(new_input)-3, 2) :
    if new_input[i] == '-' :
        new_input[i+2] = '-'

count = int(new_input[0])
for i in range(1, len(new_input)-1, 2) :
    if new_input[i] == '-' :
        count -= int(new_input[i+1])
    elif new_input[i] == '+' :
        count += int(new_input[i+1])

print(count)

File: 백준 문제 풀이 모음/kotlin/class 3/1541.kt
package class3

fun main() =with(System.`in`.bufferedReader()) {
    val str = readLine().split('-').toList()


//    print(str)

    var result2 = 0
    var count = 0
    str.forEach {
        count ++
        var result = 0
        val cost = it.split('+').toList()
        cost.forEach {
            result += it.toInt()
        }
        if (count == 1) {
            result2 += result
        } else {
            result2 -= result
        }
    }


    println(result2)


}